hw_3/hw_3.py

```diff
@@ -0,0 +1,78 @@
+# Description
+# homEwork: tHis iz your homeWork, copy these Text to variable.
+#
+# You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each
+# existING SENtence and add it to the END OF this Paragraph.
+#
+#   it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+#
+# last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, but ALL whitespaces. I
+# got 87.
+
+
+string_value = """  
+homEwork:
+    tHis iz your homeWork, copy these Text to variable.
+
+    You NEED TO normalize it fROM letter CASEs point oF View. also, create one MORE senTENCE witH LAST WoRDS of each existING SENtence and add it to the END OF this Paragraph.
+
+    it iZ misspeLLing here. fix“iZ” with correct “is”, but ONLY when it Iz a mistAKE.
+
+    last iz TO calculate nuMber OF Whitespace characteRS in this Tex. caREFULL, not only Spaces, but ALL whitespaces. I got 87.
+
+"""
+
+counter = 0     # variable to count whitespaces
+list_of_last_words = []     # variable to keep list of last words in sentences
+flag_to_change_case = False     # flag to detect first word in sentence
+new_string = ''     # variable to keep last words of each sentence
+
+string_value = string_value.lower()     # converting string to lover case
+string_value = string_value.capitalize()  # converting string to make first word capitalized
+split_string = string_value.split()     # variable to keep list of words of the provided text
+
+# normalizing string from letter case point of view
+for x in range(len(string_value) - 2):
+    # check flag and that current element is a char
+    if flag_to_change_case and string_value[x].isalpha():
+        # changing case of found char
+        string_value = string_value[0:x] + string_value[x].swapcase() + string_value[x + 1: len(string_value)]
+        flag_to_change_case = False            # changing back flag value
+    if string_value[x] in ['.', '\n', '\t']:   # check to detect end of sentence to change flag value
+        flag_to_change_case = True             # changing flag value
+
+# detecting last words in each string
+for x in range(len(split_string)):
+    if split_string[x][len(split_string[x]) - 1] == '.':    # check if split word contains dot at the end
+        list_of_last_words.append(split_string[x][:len(split_string[x]) - 1])   # adding word to list
```

Comment:
I propose to split on sentences using regular expressions and find last words in sentences also using regular expression.